frontend/javascripts/oxalis/view/right-border-tabs/starting_job_modals.tsx

```diff
@@ -1,78 +1,268 @@
+          marginRight: 6,
+        }}
+      />
+      {bbox.name} ({computeArrayFromBoundingBox(bbox.boundingBox).join(", ")})
+    </>
+  );
+}
 
-    if (selectedBBox) {
-      setSelectedBoundingBox(selectedBBox);
-    }
-  };
+function BoundingBoxSelectionFormItem({
+  isBoundingBoxConfigurable,
+  userBoundingBoxes,
+}: BoundingBoxSelectionProps): JSX.Element {
+  return (
+    <div style={isBoundingBoxConfigurable ? {} : { display: "none" }}>
+      <p>
+        Please select the bounding box for which the inferral should be computed. Note that large
+        bounding boxes can take very long. You can create a new bounding box for the desired volume
+        with the bounding box tool in the toolbar at the top. The created bounding boxes will be
+        listed below.
+      </p>
+      <Form.Item
+        label="Bounding Box"
+        name="boundingBoxId"
+        rules={[
+          {
+            required: isBoundingBoxConfigurable,
+            message: "Please select the bounding box for which the inferral should be computed.",
+          },
+        ]}
+        hidden={!isBoundingBoxConfigurable}
+      >
+        <Select
+          showSearch
+          placeholder="Select a bounding box"
+          optionFilterProp="children"
+          filterOption={(input, option) =>
+            // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
+            option.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
+          }
+        >
+          {userBoundingBoxes.map((userBB) => (
+            <Select.Option key={userBB.id} value={userBB.id}>
+              {renderUserBoundingBox(userBB)}
+            </Select.Option>
+          ))}
+        </Select>
+      </Form.Item>
+    </div>
+  );
+}
+
+type OutputSegmentationLayerNameProps = {
+  hasOutputSegmentationLayer: boolean;
+  layers: APIDataLayer[];
+  additionalAllowedNames: string[];
+};
+
+export function OutputSegmentationLayerNameFormItem({
+  hasOutputSegmentationLayer,
+  layers,
+  additionalAllowedNames,
+}: OutputSegmentationLayerNameProps) {
+  return (
+    <Form.Item
+      label="Name of output segmentation layer"
+      name="outputSegmentationLayerName"
+      rules={[
+        { required: hasOutputSegmentationLayer },
+        {
+          min: 3,
+        },
+        {
+          pattern: /[0-9a-zA-Z_-]+$/,
+        },
+        {
+          validator: async (_rule, newOutputLayerName) => {
+            if (
+              layers.some((layer) => layer.name === newOutputLayerName) &&
+              !additionalAllowedNames.includes(newOutputLayerName)
+            ) {
+              const reason =
+                "This name is already used by another segmentation layer of this dataset.";
+              return Promise.reject(reason);
+            } else {
+              return Promise.resolve();
+            }
+          },
+        },
+      ]}
+      hidden={!hasOutputSegmentationLayer}
+    >
+      <Input />
+    </Form.Item>
+  );
+}
+
```

Comment:
This new `Form.Item` is new and wasn't reviewed by @philippotto. Could you please do that @daniel-wer?

Reply:
Could you explain the rationale behind `additionalAllowedNames`? I didn't understand why this is needed from looking at the code.

Reply:
`additionalAllowedNames` is a list of names that are allowed to enter although the dataset already has a layer with that name. In the modal this is used to allow the user to let the output segmentation layer have the exact same name as the input segmentation layer. This is possible because the output segmentation layer replaces the input segmentation layer in the output dataset. Therefore non `_corrected` or something like this at the end of the output layer name is needed.

Does this behaviour make sense to you?

Reply:
Ah yes that makes sense. One could also remove the extra parameter `additionalAllowedNames` and filter the `layers` before handing them as a prop to `OutputSegmentationLayerNameFormItem` (to filter out the segmentation layer that is replaced). A small comment could be added at the filtering spot to explain why this is done.
I feel like this would be cleaner and easier to understand, what do you think?

Reply:
> what do you think?

Sounds good to me :)